qa/rpc-tests/test.py

```diff
@@ -0,0 +1,235 @@
+    'blockchain.py',
+    'disablewallet.py',
+    'zcjoinsplit.py',
+    'zcjoinsplitdoublespend.py',
+    'zkey_import_export.py',
+    'reorg_limit.py',
+    'getblocktemplate.py',
+    'bip65-cltv-p2p.py',
+    'bipdersig-p2p.py',
+    'p2p_nu_peer_management.py',
+    'rewind_index.py',
+    'p2p_txexpiry_dos.py',
+    'p2p_txexpiringsoon.py',
+    'p2p_node_bloom.py',
+    'regtest_signrawtransaction.py',
+    'finalsaplingroot.py'
+)
+
+EXTENDED_SCRIPTS=(
+    'getblocktemplate_longpoll.py',
+    'getblocktemplate_proposals.py',
+    'pruning.py',
+    'forknotify.py',
+    'hardforkdetection.py',
+    'invalidateblock.py',
+    'keypool.py',
+    'receivedby.py',
+    'rpcbind_test.py',
+#   'script_test.py',
+    'smartfees.py',
+    'maxblocksinflight.py',
+    'invalidblockrequest.py',
+#    'forknotify.py',
+    'p2p-acceptblock.py'
+)
+
+NON_SCRIPTS =(
+    'rpc-tester.py'
+)
+
+def cleanup_failed_tests():
+    pass
+
+def run_test_scripts(test_list, args):
+    fail_list = []
+    pass_list = []
+    stats = []
+
+    total_tests = len(test_list)
+    num_tests_ran = 0
+
+    logging.info("=== Starting RPC Tester  ===")
+    total_stime = time.time()
+    for file in test_list:
+        logging.info("--- Running test : %s ---", file)
+        stats.append(file)
+        #might want to add routine to properly clean SIGTERM tests so they don't conga line the rest
+        try:
+            start_time = time.time()
+            p1 = Popen(['python', file])
+            buffer = p1.communicate()[0]
+        except:
+            fail_list.append(file)
+        
+        elapsed_time = time.time() - start_time
+        logging.info("Elapsed time: %s", time.strftime("%H:%M:%S", time.gmtime(elapsed_time)))
+        stats.append(time.strftime("%H:%M:%S", time.gmtime(elapsed_time)))
+       
+        if p1.returncode != 0:
+            fail_list.append(stats)
+        else:
+            pass_list.append(stats)
+        
+        num_tests_ran += 1
+
+        if args.failfast and len(fail_list) > 0 :
+            break
+    total_elapsed_time = time.time() - total_stime
+
+    logging.info("=====")
+    logging.info("Total tests: %i ", total_tests)
+    logging.info("Tests ran: %i ", num_tests_ran)
+    logging.info("Duration: ~%s ", time.strftime("%H:%M:%S", time.gmtime(total_elapsed_time)))
+    logging.info("Num Fail: %i", len(fail_list) )
+    logging.info("Num Pass: %i", len(pass_list) )
+    logging.info("=====")
+    #add sort routine
+    #maybe cache meta stuff
+    #sanity check failed tests (e.g. clean cache, kill zombie zcashd pids, dump mem info, retry 3 times)
+    logging.info("+ TEST(S) PASSED :")
+    logging.info("------------------")
+    print_test_results(pass_list)
+    logging.info("- TEST(S) FAILED :")
+    logging.info("------------------")
+    print_test_results(fail_list)
+
+def print_test_results(test_list):
+    i =0
+    for test in test_list:
+        logging.info("%s : %s", test[i], test[i+1])
```

Comment:
Why is this printing by twos?

Reply:
Str4d is correct about the counting by twos, and the issues with failed/interrupted test scenarios. Looking to address this in next round of commits.